hoodie-client/src/main/java/com/uber/hoodie/func/payload/GenericRecordBufferedIteratorPayloadFunction.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (c) 2018 Uber Technologies, Inc. (hoodie-dev-group@uber.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *          http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.uber.hoodie.func.payload;
+
+import org.apache.avro.generic.GenericRecord;
+
+/**
+ * A GenericRecord based {@link BufferedIteratorPayloadFunction}
+ */
+public class GenericRecordBufferedIteratorPayloadFunction
+    implements BufferedIteratorPayloadFunction<GenericRecord, AbstractBufferedIteratorPayload> {
+
+  @Override
+  public AbstractBufferedIteratorPayload apply(GenericRecord t) {
```

Comment:
This class for e.g is why I strongly feel we should merge payload and transform into a single entity. We always use them to together and end up with a more complex Generic interfaces and micro implementations in the current way

Reply:
in any case, lets name this  class shorter? the Transform rename would make this `GenericRecordBufferedIteratorTransform` which is a tad shorter.

Reply:
Addressed rename